GliaWidgets/Lib/Extensions/UIViewController+Extensions.swift

```diff
@@ -8,3 +8,47 @@ extension UIViewController {
         viewController.didMove(toParent: self)
     }
 }
+
+extension UIViewController {
+    private final class DeallocatingProxy {
+        var deallocating: (() -> Void)?
+
+        deinit {
+            deallocating?()
+        }
+    }
+
+    private enum AssociatedKeys {
+        static var deallocatingProxy = "deallocatingProxy"
+    }
+
+    var deallocating: (() -> Void)? {
```

Comment:
I'm sorry, but I missed a little bit the reason of using associated objects. Could you clarify this a little bit? Do we have any alternatives for it?

Reply:
Explained in Slack